x/lockup/keeper/synthetic_lock_test.go

```diff
@@ -0,0 +1,311 @@
+	suite.Require().Equal(accountUnlockableCoins.String(), "")
+	accountUnlockingCoins := suite.app.LockupKeeper.GetAccountUnlockingCoins(suite.ctx, addr1)
+	suite.Require().Equal(accountUnlockingCoins.String(), "")
+	accountLockedCoins := suite.app.LockupKeeper.GetAccountLockedCoins(suite.ctx, addr1)
+	suite.Require().Equal(accountLockedCoins.String(), "10stake")
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTime(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTimeNotUnlockingOnly(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountUnlockedBeforeTime(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 0)
+
+	err = suite.app.LockupKeeper.CreateSyntheticLockup(suite.ctx, 1, "stakedtovalidator1", false)
+	suite.Require().NoError(err)
+
+	synthLock, err := suite.app.LockupKeeper.GetSyntheticLockup(suite.ctx, 1, "stakedtovalidator1")
+	suite.Require().NoError(err)
+	suite.Require().Equal(*synthLock, types.SyntheticLock{
+		LockId:  1,
+		Suffix:  "stakedtovalidator1",
+		EndTime: time.Time{},
+	})
+
+	synthLocks := suite.app.LockupKeeper.GetAllSyntheticLockupsByLockup(suite.ctx, 1)
+	suite.Require().Len(synthLocks, 1)
+	suite.Require().Equal(*synthLock, synthLocks[0])
+
+	synthLocks = suite.app.LockupKeeper.GetAllSyntheticLockups(suite.ctx)
+	suite.Require().Len(synthLocks, 1)
+	suite.Require().Equal(*synthLock, synthLocks[0])
+
+	// check accumulation store is correctly updated for synthetic lockup
+	accum = suite.app.LockupKeeper.GetPeriodLocksAccumulation(suite.ctx, types.QueryCondition{
+		LockQueryType: types.ByDuration,
+		Denom:         "stakestakedtovalidator1",
+		Duration:      time.Second,
+	})
+	suite.Require().Equal(accum.String(), "10")
+
+	// check queries for native denom after creating synthetic lockup
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTimeDenom(suite.ctx, addr1, "stake", suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountLockedDurationNotUnlockingOnly(suite.ctx, addr1, "stake", time.Second)
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountLockedLongerDurationDenom(suite.ctx, addr1, "stake", time.Second)
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetLocksPastTimeDenom(suite.ctx, "stake", suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetLocksLongerThanDurationDenom(suite.ctx, "stake", time.Second)
+	suite.Require().Len(locks, 1)
+	amount = suite.app.LockupKeeper.GetLockedDenom(suite.ctx, "stake", time.Second)
+	suite.Require().Equal(amount.String(), "10")
+
+	// check queries for synthetic denom after creating synthetic lockup
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTimeDenom(suite.ctx, addr1, "stakestakedtovalidator1", suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
```

Comment:
Lets make an `expectedLocks` variable, and then check that the returned locks is equal to what we expect? (Rather than just checking that they are of the same length?

Reply:
Sure, can add expectedLocks. There were lots of test modifications on superfluid minimum viable version and I would like to implement this comment on that PR to avoid conflicts

Reply:
Fixed as part of this commit
https://github.com/osmosis-labs/osmosis/pull/587/commits/6f793762da20c3904f639f5c5419e846e2af1db9